### ISSUE TITLE
Point 1 to point 2 cylindrical and conic areas

### ISSUE THREAD

**Issue by Crayder** [samp-incognito/samp-streamer-plugin#139]
The concept is to be able to define a cylinder area or a conic area by two points and a radius (well, two radii for the cone, where both points would have a radius).

On both shapes the two points are the axis endpoints, obviously. The radii will be around the axis. On the cylinder that will be throughout the entire axis, and the cone would have a gradient radius from one end to the other.

**Comment by Southclaw** [comment]
I suggested this a while ago but I forgot about it, it was back when I made that zipline script. A function to grab the distance to a line is still around here: https://github.com/Southclaw/Zipline/blob/master/Zipline.inc#L317-L319 but obviously that only works for constant-diameter tube shapes not cone shapes.

**Comment by Crayder** [comment]
Well I have PAWN code for checking points in cylinders and cone defined as I suggested, I made it a while back.

And yeah I know you suggested the cylinder part once. I'm even pretty sure I commented on it.

**Comment by samp-incognito** [comment]
I do like the idea. I suppose that you'd really only need the cone with two radii, though (that would be a truncated cone or [conical frustum](http://mathworld.wolfram.com/ConicalFrustum.html), technically). If both radii are the same, it would just be a cylinder. However, in terms of area detection, I can't think of many reasons why someone would need two different radii. A simple cylinder defined along an arbitrary axis (which is effectively just a "line" with some thickness) would be far more useful, I think.

**Comment by Crayder** [comment]
> I suppose that you'd really only need the cone with two radii

Which is actually how I do it in the code I linked above...

> However, in terms of area detection, I can't think of many reasons why someone would need two different radii.

I can think of a few reasons why I alone need a cone area. One for example, my proximity mines. They check if players are in a cone. My laser tripper mines use cylinders, since a laser is just a line.

**Comment by Patrik356b** [comment]
@Crayder Spheres...

**Comment by Crayder** [comment]
@Patrik356b streamer already has spheres... lol

**Comment by Patrik356b** [comment]
nono. I mean you could use them for proximity mines and such.

**Comment by Crayder** [comment]
@Patrik356b Nope, that's a horrible concept. That's not how a proximity mine or a laser tripping mine works.

**Comment by Patrik356b** [comment]
Not horrible. Different perhaps. I don't see why you couldn't just use a
cylinder if a sphere wouldn't be adequate.

(but a proximity mine could technically be built to trigger over a
spherical area)
